src/mc-fle2-payload-iev-v2.c

```diff
@@ -376,11 +376,18 @@ bool mc_FLE2IndexedEncryptedValueV2_get_edge(const mc_FLE2IndexedEncryptedValueV
         return false;
     }
 
-    if (iev->type != kFLE2IEVTypeRangeV2) {
+    if (iev->type != kFLE2IEVTypeRangeV2 && iev->type != kFLE2IEVTypeText) {
         CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_get_edge must be called with type range");
         return false;
     }
 
+    if (iev->type == kFLE2IEVTypeRangeV2) {
```

Comment:
nit, but i think we can skip this check, since we generally assume that the IEV passed in is well-formed, and both _parse and _validate ensure that this is true.